pkg/httpclient/client_test.go

```diff
@@ -73,24 +77,61 @@ func TestClient_Do(t *testing.T) {
 
 	require.NotNil(t, resp, "returned response should not be nil")
 
-	l, err := logging.NewLogger(logging.WithLevel(zapcore.DebugLevel))
-	require.NoError(t, err, "failed creating logger: %v", err)
+	// Create a sink instance, and register it with zap for the "memory" protocol.
+	sink := &MemorySink{new(bytes.Buffer)}
+	err = zap.RegisterSink("memdiff", func(*url.URL) (zap.Sink, error) {
+		return sink, nil
+	})
+	require.NoError(t, err)
+
+	l, err := logging.NewLogger(
+		logging.WithFields(
+			zap.String("program", "test_log"),
+			zap.String("version", "1.2.3"),
+			zap.String("release", "4"),
+		),
+		logging.WithFormatStr("json"),
+		logging.WithLevelStr("debug"),
+		logging.WithOutputPaths([]string{"memdiff://"}),      // Redirect all messages to the MemorySink.
+		logging.WithErrorOutputPaths([]string{"memdiff://"}), // Redirect all errors to the MemorySink.
+	)
+	require.NoError(t, err)
+	require.NotNil(t, l)
 
 	ctx = logging.WithLogger(ctx, l)
 	req, err = http.NewRequestWithContext(ctx, http.MethodGet, server.URL, nil)
-	require.NoError(t, err, "failed creating http request with context: %v", err)
+	require.NoError(t, err)
 
 	resp, err = client.Do(req)
-	require.NoError(t, err, "client.Do() with context unexpected error = %v", err)
+	require.NoError(t, err)
 
 	defer func() {
 		err := resp.Body.Close()
-		require.NoError(t, err, "error closing resp.Body")
+		require.NoError(t, err)
 	}()
 
-	require.NotNil(t, resp, "returned response should not be nil")
+	require.NotNil(t, resp)
 
 	responseBody, err := io.ReadAll(resp.Body)
-	require.NoError(t, err, "failed reading the body content: %v", err)
+	require.NoError(t, err)
 	require.Equal(t, body, responseBody)
+
+	// check logs
+	out := sink.String()
+	require.NotEmpty(t, out, "captured log output")
+	require.Contains(t, out, `"traceid"`)
+	require.Contains(t, out, `"request":"GET / HTTP/1.1`)
+	require.Contains(t, out, `"response":"HTTP/1.1 200 OK`)
+	require.Contains(t, out, bodyStr)
```

Comment:
If possible, would it help to adjust this on to table testing, so we can also test the `logPrefix` functionality, which is not currently covered?